stdlib/buffer.gr

```diff
@@ -376,6 +376,68 @@ provide let addString = (string, buffer) => {
   buffer.len += bytelen
 }
 
+/**
+ * Gets the UTF-8 encoded character at the given byte index.
+ *
+ * @param index: The byte index to access
+ * @param buffer: The buffer to access
+ * @returns A character starting at the given index
+ *
+ * @throws IndexOutOfBounds: When `index` is negative
+ * @throws IndexOutOfBounds: When `index + 1` is greater than the bytes size
```

Comment:
It's not just + 1, right? If you ask for a character that's 4 bytes long then that whole range needs to be valid.

Reply:
So its `index + 1` to get the usv and if the length doesn't match up then its just an invalid usv.

Reply:
```suggestion
 * @throws IndexOutOfBounds: When `index + 1` is greater than the buffer size
```